### ISSUE TITLE
Color Format Doesn't save? 

### ISSUE THREAD

**Issue by EnderProjects** [EbonJaeger/perworldinventory-kt#176]
**Describe the bug**
Upon entering and leaving new worlds, items with custom name and/or lore will lose their format code and be replaced with "??6" as an example. This doesn't happen 100% of the time but due to how the server I'm apart of uses custom items for villager trading become useless. I really have no knowledge onto why it could be happening but I'm willing to send any information to help solve this issue  

**To Reproduce**
1. warp to a different world
2. Go back to original world 
3. See error
https://imgur.com/a/gHJ6MjA

**Expected behavior**
Format to stay
https://imgur.com/a/Z4wbJNd

**Plugin version**
- [2.3.2 ] Using the latest released version of the plugin

**Server version**
1.15.2

**Comment by FamiliaBadilia** [comment]
I'm having the same problem, if you find a fix could you please let me know?

**Comment by EbonJaeger** [comment]
This would be a bug with the server software. I think I remember seeing reports of this with Spigot or Paper, not sure which.

**Comment by EnderProjects** [comment]
must be spigot then I'll look around there thank you!

**Comment by EbonJaeger** [comment]
Link to Spigot bug: https://hub.spigotmc.org/jira/browse/SPIGOT-5063

**Comment by EnderProjects** [comment]
As i was testing a new command I noticed that items loss data upon cache

# Set the duration in minutes for player profile information to be cached
cache-duration: 10

If I were to go back to a world before that time is up. it seems to not break the items

**Comment by EbonJaeger** [comment]
@EnderProjects yes, because that is being cached in memory and thus does not have to be deserialized by the server, which breaks it.

**Comment by EbonJaeger** [comment]
I'm looking at creating a workaround for this issue, but colors save properly for me on latest Paper. Can anyone verify that this works for them? Just rename this file from `.zip` to `.jar`:
[perworldinventory-kt-2.3.3.zip](https://github.com/EbonJaeger/perworldinventory-kt/files/4589061/perworldinventory-kt-2.3.3.zip)

**Comment by EnderProjects** [comment]
We have moved to paper since recently and haven't got the same issue yet. Is it safe to use the jar you provided in this case?

**Comment by EbonJaeger** [comment]
@EnderProjects it should be, yeah. All it should be doing is going through all of the color codes and replacing any broken ones.

**Comment by EnderProjects** [comment]
Ok, so after a server restart. It seemed to of happen right when I logged on 
https://imgur.com/a/NULoT55
After that I tested going to and from worlds, resulting in the same outcome with a reload in between.
ex: tp to world, reload, tp back 
https://imgur.com/a/ayk96E5

**Comment by EbonJaeger** [comment]
Can you send me the json file for the player data that it happens on?

**Comment by EnderProjects** [comment]
https://pastebin.com/nCsxsQLK

**Comment by EnderProjects** [comment]
Also by using the method I said before by tp, reload, tp has a 100% so far on recreating the issue.

**Comment by jk526** [comment]
Is there any update on this? We're having the exact same issue!

**Comment by Ash10744** [comment]
Having this issue on version 1.16.4
![unknown-19](https://user-images.githubusercontent.com/16086663/104547617-df4c5600-5626-11eb-95d1-9c020d235485.png)
